### ISSUE TITLE
improve docs for a broader audience

### ISSUE THREAD

**Issue by patch** [patch/cldr-number-pm5#51]
**tl;dr: Let’s improve the docs! Please add doc requests or suggestions in the comments here.**

The first goal of this project was to implement the standardized Unicode CLDR–based localized number formatting defined in [UTS #35, Part 3: Numbers](http://www.unicode.org/reports/tr35/tr35-numbers.html). Much of the CLDR::Number documentation, however, does not go into detail to describe functionality to developers without existing familiarity with the CLDR. This project shouldn’t require external knowledge in order to use it. One problem is that it allows for a lot of advanced customization that most developers will never need to use or know about when they can instead depend on the defaults provided for the requested locale (and currency for prices). Perhaps the docs should be split into 100% self-contained intro-level with more examples, and advanced-level with all the gritty options and external references. These days I write much more documentation for developers than actual coding, and while I have less time for maintaining my CPAN modules, I’d like to commit some time to improve these docs.

Thanks to @Ovid for bringing this to my attention:
- “[I'd love to see explanation of what the rounding_increment means and some examples :)](https://twitter.com/OvidPerl/status/742335521364578304)”
- “[Oh, and a link to the source data so I could understand it better! Thanks.](https://twitter.com/OvidPerl/status/742335655959793666)”
